src/plugins/dashboard/public/dashboard_app/url/sync_dashboard_url_state.ts

```diff
@@ -106,3 +107,28 @@ export const startSyncingDashboardUrlState = ({
   const stopWatchingAppStateInUrl = () => appStateSubscription.unsubscribe();
   return { stopWatchingAppStateInUrl };
 };
+
+export const startSyncingExpandedPanelState = ({
+  dashboardAPI,
+  history,
+}: {
+  dashboardAPI: DashboardAPI;
+  history: History;
+}) => {
+  const expandedPanelSubscription = dashboardAPI?.expandedPanelId.subscribe(() => {
+    const expandedPanel = dashboardAPI.expandedPanelId.value;
+    if (expandedPanel) {
+      history.replace({
+        ...history.location,
+        pathname: `${createDashboardEditUrl(dashboardAPI.savedObjectId.value)}/${expandedPanel}`,
+      });
+    } else {
+      history.replace({
+        ...history.location,
+        pathname: createDashboardEditUrl(dashboardAPI.savedObjectId.value),
+      });
+    }
```

Comment:
Can we use a ternary here so we only have one call to `history.replace`? i.e. call `history.replace` once but choose how to overwrite `pathname` dependendent on if there is an expanded panel ID.

Reply:
Sweet! [0a89da4](https://github.com/elastic/kibana/pull/190086/commits/0a89da4d9bd232b585f0b560881ee3b6abad367c) for changes!